---
title: "Tracking an ecosystem shift, from microorganisms to mammals, with environmental DNA"
author: "Anni Djurhuus"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(include = FALSE)
```

```{r, include=FALSE}
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7","red","grey", "grey20")

add.alpha <- function(col, alpha=1){
  if(missing(col))
    stop("Please provide a vector of colours.")
  apply(sapply(col, col2rgb)/255, 2, 
                     function(x) 
                       rgb(x[1], x[2], x[3], alpha=alpha))  
}
cbbPalette_alpha <- add.alpha(cbbPalette, alpha=0.5)
```

##### To do list:
4. Put all genes into the same dataset, resulting in 2 datasets (one per geographic location), each of which has families in rows, samples in columns.
6. Look for wholesale shifts over the 2015-2016 timeframe. Anchovies-sardines is the big one I’ve been suspecting. This may or may not be related to El Niño.
7. Correlate data with Anchovies (on ranks, i.e. spearman or kendell), not temperature. Make a correlation matrix.

8. Filter OTUs from occupancy data
9. Decontaminate with Ryans script
10. Eveness of communities with samples. 

General guidelines for Nature manuscripts:
-Manuscript do not normally exceed 5 pages of Nature
-Up to 50 references (One page of undiluted text is about 1,300 words.)
-Summary up to 150 words: This summary contains a paragraph (2-3 sentences) of basic-level introduction to the field; a brief account of the background and rationale of the work; a statement of the main conclusions (introduced by the phrase 'Here we show' or its equivalent); and finally, 2-3 sentences putting the main findings into general context so it is clear how the results described in the paper have moved the field forwards.
-main text of no more than 3,500 words and 6 display items (figures, tables).
-Statistical information: Comprehensive information on the statistical analyses used must be included in the paper. The Methods must include a statistics section where you describe the statistical tests used and whether they were one- or two-tailed. Please ensure that the error bars are defined throughout the figures. For all statistics (including error bars), provide the EXACT n values used to calculate the statistics. Where relevant, provide exact values for both significant and non-significant P values. For ANOVAs, provide F values and degrees of freedom. For t-tests, provide t-values and degrees of freedom. Please specifically define the replicates.

```{r load libraries, include=FALSE}
#install.packages("googledrive")
library("googledrive")
library(tibble)
library("magrittr")
library("phyloseq")
library("ggplot2")
library("reshape2")
library(cowplot)
library(ggplot2)
library(lubridate)
```

###Main 
Global change in climate, environments, and shifts in biodiversity distribution necessitate assessments to provide a baseline measure to gauge undergoing changes. Traditionally marine biodiversity assessments are conducted via trawling or aerial surveys focusing on macro-organisms, most commonly those species that have a market value. More recently the global and local assessments have been conducted to explore microbial diversity unveiling the vast microbial diversity that exists. Cross trophic level assessments of biodiversity are rarely conducted, and the simultaneous biodiversity fluxes from microorganisms to macro-organisms over temporal scales are non-existing, though it is with time-series studies where changes in the biodiversity can be assessed. Whether due to anthropogenic or natural disturbances there is a general trend of biodiversity decline (Butchart et al, 2010). Efforts to assess these changes and declines have not yet connected the trophic level interactions to understand the total community impact. In order to make biodiversity assessments more accurate, we require better methods and increased observation efforts of all organisms. 

While climate and aspects of environmental change can be measured autonomously, biodiversity assessments have traditionally been conducted manually. The genetic trace by the cell(s) or extracellular DNA of an organism in the surrounding environment, referred to as environmental DNA (eDNA), allows for the detection of that organism from an environmental sample. This method has been applied in microbial ecology in many terrestrial and aquatic studies and has been more recently applied to detection of eukaryotic organisms. These methods are also being adapted and applied to detection of cancer types in human blood samples (Diaz & Bardelli, 2014). As these methods become more optimized, the molecular detection of organisms has the potential to assist in environmental health assessments and management applications.

Sequencing of environmental samples such as seawater is now applied in a macrobial sense across a range of trophic levels (“microbes to mammals”) (Trivedi et al., 2016; Valentini et al., 2016), and approaches used to inventory individuals (Ardura et al., 2013; Harada et al., 2015; Thompson et al., 2016) are being applied to assemblages (Carugati et al., 2015; Zimmermann et al., 2015; Aylagas et al., 2016, Djurhuus et al., 2018). Studies demonstrate DNA metabarcoding to be a reliable method for biodiversity assessment with potential for inferring biotic indices for marine ecosystem quality assessment (Aylagas et al., 2014; Pawlowski et al., 2014; Cowart et al., 2015; Elbrecht and Leese, 2015; Visco et al., 2015; Ferrera et al., 2016).

Increased monitoring efforts and leveraging eDNA to inform marine biodiversity trends are goals of the Marine Biodiversity Observation Network (MBON) (Muller-Karger et al., 2014). This initiative has tried to address gaps in the marine realm, where monitoring has been standard in many terrestrial environments (Duffy et al, 2013). 

Here we explore the ability of multiple markers to detect organisms in water column samples from bacteria to vertebrates. Our study detects seasonal trends as well as changes in biodiversity during the 2015-2016 El Niño event. 

###Methods
All data was agglomerated by family and all annotations representing fewer than 0.001% of the data were filtered out.  

All analysis was done using the R statistical software (R Core Team, 2018). 

```{r read files in and subset to appropriate samples FK, include=FALSE}
FK_16S_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/M2W_16S_FK_json_obs_md-2.biom",parseFunction = parse_taxonomy_default)
FK_18S_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/M2W_18S_FK_json_obs_md.biom",parseFunction = parse_taxonomy_default)
FK_COI_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/M2W_COI_FK_json_obs_md.biom",parseFunction = parse_taxonomy_default)
FK_12S_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/FK_12S_NT_json_obs_md.biom",parseFunction = parse_taxonomy_default)

FK_16S <- merge_phyloseq(FK_16S_biom@otu_table, FK_16S_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_16S_FK_sam_data.csv", row.names=2)))
FK_18S <- merge_phyloseq(FK_18S_biom@otu_table, FK_18S_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_18S_FK_sam_data.csv", row.names=2)))
FK_COI <- merge_phyloseq(FK_COI_biom@otu_table, FK_COI_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_COI_FK_sam_data.csv", row.names=2)))
FK_12S <- merge_phyloseq(FK_12S_biom@otu_table, FK_12S_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_12S_FK_sam_data.csv", row.names=2)))

FK_16S <- subset_samples(FK_16S, description == "m2w")
FK_16S_prop  = transform_sample_counts(FK_16S, function(x) x / sum(x))
FK_16S_prop = filter_taxa(FK_16S, function(x) sum(x) > .001, TRUE)

FK_18S <- subset_samples(FK_18S, description == "m2w")
FK_18S_prop  = transform_sample_counts(FK_18S, function(x) x / sum(x))
FK_18S_prop = filter_taxa(FK_18S, function(x) sum(x) > .001, TRUE)

FK_COI <- subset_samples(FK_COI, description == "m2w")
FK_COI_prop  = transform_sample_counts(FK_COI, function(x) x / sum(x))
FK_COI_prop = filter_taxa(FK_COI, function(x) sum(x) > .001, TRUE)

FK_12S <- subset_samples(FK_12S, description == "m2w")
FK_12S_prop  = transform_sample_counts(FK_12S, function(x) x / sum(x))
FK_12S_prop = filter_taxa(FK_12S, function(x) sum(x) > .001, TRUE)
```

```{r read files in and subset to appropriate samples MB, include=FALSE}
MB_16S_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/M2W_16S_MB_json_obs_md.biom",parseFunction = parse_taxonomy_default)
sample_names(MB_16S_biom)
MB_18S_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/M2W_18S_MB_json_obs_md.biom",parseFunction = parse_taxonomy_default)
sample_names(MB_18S_biom)
MB_COI_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/M2W_COI_MB_json_obs_md.biom",parseFunction = parse_taxonomy_default)
MB_12S_biom <- import_biom("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/biom/MB_12S_NT_json_obs_md.biom",parseFunction = parse_taxonomy_default)

MB_16S <- merge_phyloseq(MB_16S_biom@otu_table, MB_16S_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_16S_MB_metadata.csv", row.names=2)))
MB_18S <- merge_phyloseq(MB_18S_biom@otu_table, MB_18S_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_18S_MB_metadata.csv", row.names=2)))
MB_COI <- merge_phyloseq(MB_COI_biom@otu_table, MB_COI_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_COI_MB_metadata.csv", row.names=2)))
MB_12S <- merge_phyloseq(MB_12S_biom@otu_table, MB_12S_biom@tax_table,sample_data(read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/google_drive/data/MEGAN6/metadata/m2w_12S_MB_metadata.csv", row.names=2)))

MB_16S <- subset_samples(MB_16S, description == "m2w")
MB_16S_prop  = transform_sample_counts(MB_16S, function(x) x / sum(x))
MB_16S_prop = filter_taxa(MB_16S, function(x) sum(x) > .001, TRUE)

MB_18S <- subset_samples(MB_18S, description == "m2w")
MB_18S_prop  = transform_sample_counts(MB_18S, function(x) x / sum(x))
MB_18S_prop = filter_taxa(MB_18S, function(x) sum(x) > .001, TRUE)

MB_COI <- subset_samples(MB_COI, description == "m2w")
MB_COI_prop  = transform_sample_counts(MB_COI, function(x) x / sum(x))
MB_COI_prop = filter_taxa(MB_COI, function(x) sum(x) > .001, TRUE)

MB_12S <- subset_samples(MB_12S, description == "m2w")
MB_12S_prop  = transform_sample_counts(MB_12S, function(x) x / sum(x))
MB_12S_prop = filter_taxa(MB_12S, function(x) sum(x) > .001, TRUE)
```

```{r agglomerate by family FK, include=FALSE}
FK_16S_fam <- tax_glom(FK_16S_prop, "Rank5")
FK_18S_fam <- tax_glom(FK_18S_prop, "Rank5")
FK_COI_fam <- tax_glom(FK_COI_prop, "Rank5")
FK_12S_fam <- tax_glom(FK_12S_prop, "Rank5")
```

```{r agglomerate by family MB, include=FALSE}
MB_16S_fam <- tax_glom(MB_16S_prop, "Rank5")
MB_18S_fam <- tax_glom(MB_18S_prop, "Rank5")
MB_COI_fam <- tax_glom(MB_COI_prop, "Rank5")
MB_12S_fam <- tax_glom(MB_12S_prop, "Rank5")
```


###Results
####Plots
Richness calculated on observed number of families. Only data with a relative abundance above 0.001% is included.
```{r plotting richness based on family FK, echo=TRUE}
richness_16S_FK <- setNames(cbind(estimate_richness(FK_16S_fam, measures="Observed"),FK_16S@sam_data$collection_date, FK_16S@sam_data$locus),c("richness","YEAR_DATE", "locus"))
richness_18S_FK <- setNames(cbind(estimate_richness(FK_18S_fam, measures="Observed"),FK_18S@sam_data$collection_date, FK_18S@sam_data$locus),c("richness","YEAR_DATE", "locus"))
richness_COI_FK <- setNames(cbind(estimate_richness(FK_COI_fam, measures="Observed"),FK_COI@sam_data$collection_date, FK_COI@sam_data$locus),c("richness","YEAR_DATE", "locus"))
richness_12S_FK <- setNames(cbind(estimate_richness(FK_12S_fam, measures="Observed"),FK_12S@sam_data$collection_date, FK_12S@sam_data$locus),c("richness","YEAR_DATE", "locus"))

#par(mar=c(1,1,3,4))
boxplot(richness_16S_FK$richness/max(richness_16S_FK$richness)~ymd(richness_16S_FK$YEAR_DATE), at=c(1,3,6,8,10,12,14,16,18,20),xaxt="n",main="FKNMS", cex.axis = 0.8,las=2,ylim=c(0,1.1), yaxt="n",col=cbbPalette_alpha[2])
par(new=TRUE)
boxplot(richness_18S_FK$richness/max(richness_18S_FK$richness)~ymd(richness_18S_FK$YEAR_DATE), at=c(1,3,6,8,10,12,14,16,18,20),ylim=c(0,1.1),xaxs="i", yaxt="n",xaxt="n", col=cbbPalette_alpha[3],alpha=0.4)
par(new=TRUE)
boxplot(richness_COI_FK$richness/max(richness_COI_FK$richness)~ymd(richness_COI_FK$YEAR_DATE), at=c(1,3,6,8,10,12,14,16,18,20), ylim=c(0,1.1),yaxt="n",xaxt="n",col=cbbPalette_alpha[4],alpha=0.5)
par(new=TRUE)
boxplot(richness_12S_FK$richness/max(richness_12S_FK$richness)~ymd(richness_12S_FK$YEAR_DATE), at=c(1,3,6,8,10,12,14,16,18,20), ylim=c(0,1.1),yaxt="n",xaxt="n",col=cbbPalette_alpha[11],alpha=0.9)
legend("bottomright",title="Locus",c("16S - microorganisms","18S - phytoplankton","COI - invertebrates","12S - vertebrates"), fill=c(cbbPalette_alpha[2],cbbPalette_alpha[3],cbbPalette_alpha[4],cbbPalette_alpha[11]),  ncol=1,cex=0.8, bty="n")
abline(v=c(10), col=c("black"), lty=c(2), lwd=c(1))
axis(side=1, at=c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21),
     labels=c("Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec", "Jan", "Feb","Mar",
              "Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec"), las=3,cex.axis=0.7)
mtext("2015", side=1, line=3, at=4,cex=0.8)
mtext("2016", side=1, line=3, at=16,cex=0.8)
```

```{r plotting richness based on family MB, echo=TRUE}
richness_16S_MB <- setNames(cbind(estimate_richness(MB_16S_fam, measures="Observed"),MB_16S@sam_data$collection_date, MB_16S@sam_data$locus),c("richness","YEAR_DATE", "locus"))
richness_18S_MB <- setNames(cbind(estimate_richness(MB_18S_fam, measures="Observed"),MB_18S@sam_data$collection_date, MB_18S@sam_data$locus),c("richness","YEAR_DATE", "locus"))
richness_COI_MB <- setNames(cbind(estimate_richness(MB_COI_fam, measures="Observed"),MB_COI@sam_data$collection_date, MB_COI@sam_data$locus),c("richness","YEAR_DATE", "locus"))
richness_12S_MB <- setNames(cbind(estimate_richness(MB_12S_fam, measures="Observed"),MB_12S@sam_data$collection_date, MB_12S@sam_data$locus),c("richness","YEAR_DATE", "locus"))

#par(mar=c(1,1,3,4))
boxplot(richness_16S_MB$richness/max(richness_16S_MB$richness)~ymd(richness_16S_MB$YEAR_DATE), at=c(1,4,7,9,13,17,18,21),xaxt="n",main="MBNMS", cex.axis = 0.8,las=2,ylim=c(0,1.1), yaxt="n",col=cbbPalette_alpha[2])
par(new=TRUE)
boxplot(richness_18S_MB$richness/max(richness_18S_MB$richness)~ymd(richness_18S_MB$YEAR_DATE), at=c(1,4,7,9,13,17,18,21),ylim=c(0,1.1),xaxs="i", yaxt="n",xaxt="n", col=cbbPalette_alpha[3],alpha=0.4)
par(new=TRUE)
boxplot(richness_COI_MB$richness/max(richness_COI_MB$richness)~ymd(richness_COI_MB$YEAR_DATE), at=c(1,4,7,9,13,17,18,21), ylim=c(0,1.1),yaxt="n",xaxt="n",col=cbbPalette_alpha[4],alpha=0.5)
par(new=TRUE)
boxplot(richness_12S_MB$richness/max(richness_12S_MB$richness)~ymd(richness_12S_MB$YEAR_DATE), at=c(4,7,9,17,18,21), xlim=c(0.5,21.5),ylim=c(0,1.1),yaxt="n",xaxt="n",col=cbbPalette_alpha[11],alpha=0.9)
legend("bottomright",title="Locus",c("16S - microorganisms","18S - phytoplankton","COI - invertebrates","12S - vertebrates"), fill=c(cbbPalette_alpha[2],cbbPalette_alpha[3],cbbPalette_alpha[4],cbbPalette_alpha[11]),  ncol=1,cex=0.8, bty="n")
abline(v=c(10), col=c("black"), lty=c(2), lwd=c(1))
axis(side=1, at=c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21),
     labels=c("Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec", "Jan", "Feb","Mar",
              "Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec"), las=3,cex.axis=0.7)
mtext("2015", side=1, line=3, at=4,cex=0.8)
mtext("2016", side=1, line=3, at=16,cex=0.8)

```

```{r simple correlations FK}
tmp_cor_16S_FK <- cor.test(richness_16S_FK$richness,as.numeric(FK_16S@sam_data$TMP_C),method=c("pearson"))
tmp_cor_18S_FK <- cor.test(richness_18S_FK$richness,as.numeric(FK_18S@sam_data$TMP_C),method=c("pearson"))
tmp_cor_COI_FK <- cor.test(richness_COI_FK$richness,as.numeric(FK_COI@sam_data$TMP_C),method=c("pearson"))
tmp_cor_12S_FK <- cor.test(richness_12S_FK$richness,as.numeric(FK_12S@sam_data$TMP_C),method=c("pearson"))

chl_cor_16S_FK <- cor.test(richness_16S_FK$richness,as.numeric(FK_16S@sam_data$CHL_MG_M.3),method=c("pearson"))
chl_cor_18S_FK <- cor.test(richness_18S_FK$richness,as.numeric(FK_18S@sam_data$CHL_MG_M.3),method=c("pearson"))
chl_cor_COI_FK <- cor.test(richness_COI_FK$richness,as.numeric(FK_COI@sam_data$CHL_MG_M.3),method=c("pearson"))
chl_cor_12S_FK <- cor.test(richness_12S_FK$richness,as.numeric(FK_12S@sam_data$CHL_MG_M.3),method=c("pearson"))
```

```{r simple correlations MB}
tmp_cor_16S_MB <- cor.test(richness_16S_MB$richness,as.numeric(MB_16S@sam_data$temp),method=c("pearson"))
tmp_cor_18S_MB <- cor.test(richness_18S_MB$richness,as.numeric(MB_18S@sam_data$temp),method=c("pearson"))
tmp_cor_COI_MB <- cor.test(richness_COI_MB$richness,as.numeric(MB_COI@sam_data$temp),method=c("pearson"))
tmp_cor_12S_MB <- cor.test(richness_12S_MB$richness,as.numeric(MB_12S@sam_data$temp),method=c("pearson"))

chl_cor_16S_MB <- cor.test(richness_16S_MB$richness,as.numeric(MB_16S@sam_data$chlorophyll),method=c("pearson"))
chl_cor_18S_MB <- cor.test(richness_18S_MB$richness,as.numeric(MB_18S@sam_data$chlorophyll),method=c("pearson"))
chl_cor_COI_MB <- cor.test(richness_COI_MB$richness,as.numeric(MB_COI@sam_data$chlorophyll),method=c("pearson"))
chl_cor_12S_MB <- cor.test(richness_12S_MB$richness,as.numeric(MB_12S@sam_data$chlorophyll),method=c("pearson"))
```

##### Correlation with environmental variables
We conducted a pearson correlation on total number of families with temperature for each genetic marker for both locations. The only two datasets that correlated with temperature were 16S from FK (p = `r tmp_cor_16S_MB[3]`) and 18S from MB (p = `r tmp_cor_18S_MB[3]`). All other datasets were as follows, Florida Keys: 18S p = `r tmp_cor_18S_FK[3]`, COI p = `r tmp_cor_COI_FK[3]`, and 12S p = `r tmp_cor_12S_FK[3]`, Monterey Bay: 16S p = `r tmp_cor_16S_MB[3]`, p = `r tmp_cor_COI_MB[3]`, and p = `r tmp_cor_12S_MB[3]`, 

In addition to temperature 12S from Monterey Bay was strongly correlated with chlorophyll (p = `r chl_cor_12S_MB[3]`)
