---
title: "Tracking ecosystem shifts, from microbes to mammals, with environmental DNA"
author: "Djurhuus et al."
output: html_document
---

```{r setup, include=FALSE}
#knitr::opts_chunk$set(include = FALSE)
library(here); library(plyr); library(biomformat); library(tidyverse); library(phyloseq); library(vegan); library(reshape2); library(MASS); library(stringr); library(purrr); library(ggplot2); library(Hmisc); library(lubridate); library(corrplot); library(gplots); library(rafalib);library(tissuesGeneExpression); library(genefilter); library(RColorBrewer); library(ggdendro);library(ComplexHeatmap); library(circlize); library(colorspace); library(GetoptLong);library(WGCNA); library(matrixStats);library(Hmisc); library(splines);library(foreach);library(doParallel);library(fastcluster); library(dynamicTreeCut); library(survival); library(WGCNA); library(cluster); library(beepr);library(bioDist); library(FactoMineR);library(MASS); library(ggfortify); library(factoextra)
```

```{r, include=FALSE}
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7","red","grey", "grey20")
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

add.alpha <- function(col, alpha=1){
  if(missing(col))
    stop("Please provide a vector of colours.")
  apply(sapply(col, col2rgb)/255, 2, 
                     function(x) 
                       rgb(x[1], x[2], x[3], alpha=alpha))  
}
cbbPalette_alpha <- add.alpha(cbbPalette, alpha=0.5)

pal <- colorRampPalette(brewer.pal(11, "RdYlGn"))(50)
```

```{r read in MB data undergone decontamination, include=FALSE}
MB_16S_m2w <- merge_phyloseq(otu_table(as.matrix(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/16S/MB_16S_otu_table.csv", row.names = 1,check.names = FALSE)), taxa_are_rows = TRUE),tax_table(as.matrix(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/16S/MB_16S_tax_table.csv", row.names = 1))),sample_data(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/16S/MB_16S_sam_table.csv", row.names = 1)))

MB_18S_m2w <- merge_phyloseq(otu_table(as.matrix(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/18S/MB_18S_otu_table.csv", row.names = 1,check.names = FALSE)), taxa_are_rows = TRUE),tax_table(as.matrix(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/18S/MB_18S_taxa_table.csv", row.names = 1))),sample_data(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/18S/MB_18S_sam_table.csv", row.names = 1)))

MB_COI_m2w <- merge_phyloseq(otu_table(as.matrix(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/COI/MB_COI_otu_table.csv", row.names = 1,check.names = FALSE)), taxa_are_rows = TRUE),tax_table(as.matrix(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/COI/MB_COI_taxa_table.csv", row.names = 1))),sample_data(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/COI/MB_COI_sam_table.csv", row.names = 1)))

MB_12S_m2w <- merge_phyloseq(otu_table(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/12S/MB_12S_otu_table.csv", row.names = 1, check.names = FALSE), taxa_are_rows = TRUE), tax_table(as.matrix(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/12S/MB_12S_taxa_table.csv", row.names = 1))), sample_data(read.csv(file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/12S/MB_12S_sam_data.csv",row.names = 1)))
```

```{r functions for splitting species, adding taxonomy to highest level available and pseudoautocorrelation, include=FALSE,message=FALSE, warning=FALSE}
split_species = function(string, n = 2) {
  splits = str_split(string, "/", n + 1)
  res = map_if(splits, ~length(.x) > 2, ~.x[1:n]) %>%
    map_chr(str_c, collapse = "/")
  return(res)
}

add_taxonomy_column = function(physeq, num_species = 2) {
  tax_df = as.data.frame(tax_table(physeq)) %>%
    rownames_to_column("OTU") %>%
    mutate(Rank7 = split_species(Rank7, n = num_species)) %>%
    mutate(Taxonomy =
             case_when(
               is.na(Rank5)  ~ str_c("o:", Rank4),
               is.na(Rank6)   ~ str_c("f:", Rank5),
               is.na(Rank7) ~ str_c("g:", Rank6)
             )
    )
  
  tax = as.matrix(tax_df[, -1])
  rownames(tax) = tax_df$OTU
  tax_table(physeq) = tax_table(tax)
  
  return(physeq)
}

add_taxonomy_column2 = function(physeq, num_species = 2) {
  tax_df = as.data.frame(tax_table(physeq)) %>%
    rownames_to_column("OTU") %>%
    mutate(Rank7 = split_species(Rank7, n = num_species)) %>%
    mutate(Taxonomy2 =
             case_when(
               is.na(Taxonomy)   ~ str_c("c:", Rank3),
               is.na(Rank6)   ~ str_c(Taxonomy)             
             )
    )
  
  tax = as.matrix(tax_df[, -1])
  rownames(tax) = tax_df$OTU
  tax_table(physeq) = tax_table(tax)
  
  return(physeq)
}

add_taxonomy_column3 = function(physeq, num_species = 2) {
  tax_df = as.data.frame(tax_table(physeq)) %>%
    rownames_to_column("OTU") %>%
    mutate(Rank7 = split_species(Rank7, n = num_species)) %>%
    mutate(Taxonomy3 =
             case_when(
               is.na(Taxonomy2)   ~ str_c("p:", Rank2),
               is.na(Rank6)   ~ str_c(Taxonomy2)             
             )
    )
  
  tax = as.matrix(tax_df[, -1])
  rownames(tax) = tax_df$OTU
  tax_table(physeq) = tax_table(tax)
  
  return(physeq)
}

PA<-function(OTUs, DATES, REFERENCE = 1, METHOD = "euclidian"){

  #library(vegan,verbose = F)
  
  #order(DATES) #make sure these are in chronological order; enforce this rule by re-ordering, like so:
  OTUs<-OTUs[,order(DATES)]
  DATES<-sort(DATES)
  
  referenceTimePoint<-unique(DATES)[REFERENCE] #set a reference time point at a particular sample
  dayVector<-as.numeric(DATES-DATES[match(referenceTimePoint, DATES)]) #time difference in days, relative to reference point
  
  bcdist<-as.matrix(vegan::vegdist(t(OTUs), method=METHOD)) 
    diag(bcdist)<-NA   #don't count self-comparison within replicates
  
    distFUN <- function(x) as.vector(bcdist[DATES==x,DATES==referenceTimePoint]) #function to pull selected distances from matrix
  
  dist.res<-lapply(X=unique(DATES), FUN=distFUN)
    #dist.res<-as.data.frame(t(plyr::ldply(dist.res, rbind)))
    	maxlength<-max(sapply(dist.res, length))		
		for(i in 1:length(dist.res)){length(dist.res[[i]])<-maxlength}	#pad to create even columns
	  dist.res<-as.data.frame(dist.res, check.names=FALSE)	
    names(dist.res)<-unique(dayVector)
    
  return(dist.res)  
  
}

OneStep<-function(OTUs, DATES, METHOD = "euclidian"){
  
  #order(DATES) #make sure these are in chronological order; enforce this rule by re-ordering, like so:
  OTUs<-OTUs[,order(DATES)]
  DATES<-sort(DATES)
  
  dayVector<-as.numeric(unique(DATES)[2:length(unique(DATES))])-
  	as.numeric(unique(DATES)[1:(length(unique(DATES))-1)]) #time difference in days, relative to previous sample

  bcdist<-as.matrix(vegan::vegdist(t(OTUs), method=METHOD)) 
    diag(bcdist)<-NA   #don't count self-comparison within replicates
  
    distFUN <- function(x) as.vector(bcdist[DATES==unique(DATES)[x], DATES==unique(DATES)[x+1]]) #function to pull selected distances from matrix

  dist.res<-lapply(1:(length(unique(DATES))-1), FUN=distFUN)
     	maxlength<-max(sapply(dist.res, length))		
		for(i in 1:length(dist.res)){length(dist.res[[i]])<-maxlength}	#pad to create even columns
	  dist.res<-as.data.frame(dist.res, check.names=FALSE)	
    names(dist.res)<-paste("Step", 1:(length(unique(DATES))-1), dayVector, "Days", sep="_")
   
  return(dist.res) 
}


multiplot <- function(..., plotlist=NULL, file, cols=1, layout=NULL) {
  library(grid)
  
  # Make a list from the ... arguments and plotlist
  plots <- c(list(...), plotlist)
  
  numPlots = length(plots)
  
  # If layout is NULL, then use 'cols' to determine layout
  if (is.null(layout)) {
    # Make the panel
    # ncol: Number of columns of plots
    # nrow: Number of rows needed, calculated from # of cols
    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                     ncol = cols, nrow = ceiling(numPlots/cols))
  }
  
  if (numPlots==1) {
    print(plots[[1]])
    
  } else {
    # Set up the page
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))
    
    # Make each plot, in the correct location
    for (i in 1:numPlots) {
      # Get the i,j matrix positions of the regions that contain this subplot
      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))
      
      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}
```

```{r transform sample abundance to proportion and make one dataframe per location, include=FALSE}
MB_16S_fam <- tax_glom(MB_16S_m2w, "Rank5")
MB_18S_fam <- tax_glom(MB_18S_m2w, "Rank5")
MB_COI_fam <- tax_glom(MB_COI_m2w, "Rank5")
MB_12S_fam <- tax_glom(MB_12S_m2w, "Rank5")

tax_table(MB_16S_fam) <- gsub('f_', as.character(NA), gsub('o_', as.character(NA), gsub('c_',as.character(NA), gsub('p_', as.character(NA),gsub('unknown', as.character(NA), tax_table(MB_16S_fam))))))
MB_16S_fam <- add_taxonomy_column(MB_16S_fam)
MB_16S_fam <- add_taxonomy_column2(MB_16S_fam)
MB_16S_fam <- subset_taxa(MB_16S_fam,Taxonomy2 !="NA")
MB_16S_fam <- tax_glom(MB_16S_fam,"Taxonomy2")
taxa_names(MB_16S_fam) <- paste(MB_16S_fam@tax_table[,9])
MB_16S_prop = transform_sample_counts(MB_16S_fam,function(x) x / sum(x))
MB_16S_prop = prune_taxa(taxa_sums(MB_16S_prop)>0, MB_16S_prop)
MB_16S_prop2 <- sweep(MB_16S_prop@otu_table,1,rowMaxs(MB_16S_prop@otu_table),"/")

#write.csv(as.data.frame(MB_16S_prop@tax_table), "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/16S/MB_16S_tax_table.csv")  #is this the best way to save biom files?

tax_table(MB_18S_fam) <- gsub('f_', as.character(NA), gsub('o_', as.character(NA), gsub('c_',as.character(NA), gsub('p_', as.character(NA), gsub('unknown', as.character(NA),tax_table(MB_18S_fam))))))
MB_18S_fam <- add_taxonomy_column(MB_18S_fam)
MB_18S_fam <- add_taxonomy_column2(MB_18S_fam)
MB_18S_fam <- subset_taxa(MB_18S_fam,Taxonomy2 !="NA")
MB_18S_fam <- tax_glom(MB_18S_fam,"Taxonomy2")
taxa_names(MB_18S_fam) <- paste(MB_18S_fam@tax_table[,9])
MB_18S_prop = transform_sample_counts(MB_18S_fam,function(x) x / sum(x))
MB_18S_prop = prune_taxa(taxa_sums(MB_18S_prop)>0, MB_18S_prop)
MB_18S_prop2 <- sweep(MB_18S_prop@otu_table,1,rowMaxs(MB_18S_prop@otu_table),"/")

#write.csv(as.data.frame(MB_18S_prop@tax_table), "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/18S/MB_18S_tax_table.csv")  #is this the best way to save biom files?

tax_table(MB_COI_fam) <- gsub('f_', as.character(NA), gsub('o_', as.character(NA), gsub('c_',as.character(NA), gsub('p_', as.character(NA), gsub('unknown', as.character(NA),tax_table(MB_COI_fam))))))
MB_COI_fam <- add_taxonomy_column(MB_COI_fam)
MB_COI_fam <- add_taxonomy_column2(MB_COI_fam)
MB_COI_fam <- subset_taxa(MB_COI_fam,Taxonomy2 !="NA")
MB_COI_fam <- tax_glom(MB_COI_fam,"Taxonomy2")
taxa_names(MB_COI_fam) <- paste(MB_COI_fam@tax_table[,9])
MB_COI_prop = transform_sample_counts(MB_COI_fam,function(x) x / sum(x))
MB_COI_prop = prune_taxa(taxa_sums(MB_COI_prop)>0, MB_COI_prop)
MB_COI_prop2 <- sweep(MB_COI_prop@otu_table,1,rowMaxs(MB_COI_prop@otu_table),"/")

#write.csv(as.data.frame(MB_COI_prop@tax_table), "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/COI/MB_COI_tax_table.csv")  #is this the best way to save biom files?

tax_table(MB_12S_fam) <- gsub('f_', as.character(NA), gsub('o_', as.character(NA), gsub('c_',as.character(NA), gsub('p_', as.character(NA), gsub('unknown', as.character(NA),tax_table(MB_12S_fam))))))
MB_12S_fam <- add_taxonomy_column(MB_12S_fam) # Add family and order assignments 
MB_12S_fam <- add_taxonomy_column2(MB_12S_fam) # Add class and phylum assignments
MB_12S_fam <- subset_taxa(MB_12S_fam,Taxonomy2 !="NA") #get rid of all NAs in dataset.
taxa_names(MB_12S_fam) <- paste(MB_12S_fam@tax_table[,9])
MB_12S_prop = transform_sample_counts(MB_12S_fam,function(x) x / sum(x))
MB_12S_prop = prune_taxa(taxa_sums(MB_12S_prop)>0, MB_12S_prop)
MB_12S_prop2 <- sweep(MB_12S_prop@otu_table,1,rowMaxs(MB_12S_prop@otu_table),"/")

#write.csv(as.data.frame(MB_12S_prop@tax_table), "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/12S/MB_12S_tax_table.csv")
#Merge all data from MB not relative abundance
MB_all_tax_ps <- merge_phyloseq(MB_COI_fam@otu_table, MB_16S_fam@otu_table,
                                MB_12S_fam@otu_table, MB_18S_fam@otu_table, MB_18S_fam@sam_data)

#Merge all OTU and taxa data from MB relative abundance. ps stands for phyloseq object, ts is for transformed
MB_all_tax_ps_ts <- merge_phyloseq(otu_table(MB_COI_prop2,taxa_are_rows = T),otu_table(MB_16S_prop2,taxa_are_rows = T), otu_table(MB_12S_prop2,taxa_are_rows = TRUE),otu_table(MB_18S_prop2,taxa_are_rows = TRUE), MB_18S_fam@sam_data) #merge data on previously transformed data.

# merge replicates for fun... 
MB_all_tax_ps_ts_merge <- merge_samples(MB_all_tax_ps_ts,"collection_date",fun=mean) #merge samples
# filter out bad taxa 
badTaxa_MB <- read.csv("/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/MB_badtaxa.csv", header=FALSE)
badTaxa_MB <- badTaxa_MB[[1]] # make bad taxa into list
goodTaxa_MB <- setdiff(taxa_names(MB_all_tax_ps_ts), badTaxa_MB) #subset good taxa
MB_all_tax_ps_rel = prune_taxa(goodTaxa_MB,MB_all_tax_ps_ts) #filter good taxa from all taxa list from all data (not merged replicates)
goodTaxa_MB_merge <- setdiff(taxa_names(MB_all_tax_ps_ts_merge), badTaxa_MB) #subset good taxa from merged replicates
MB_all_tax_ps_ts_merge = prune_taxa(goodTaxa_MB_merge,MB_all_tax_ps_ts_merge) # filter out good taxa from merged replicates
MB_merged_otu <- t(as.matrix(MB_all_tax_ps_ts_merge@otu_table)) #transpose merged otu table
#Divide all rows (taxa) with row max on transposed data
MB_merged_otu_table <- sweep(MB_merged_otu,1,rowMaxs(MB_merged_otu),"/") 
MB_all_merge <- as.data.frame(t(MB_merged_otu_table)) #create non transposed otu table

#saveRDS(MB_all_tax_ps_rel, "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/MB_all_tax_ps_rel.Rdata")  
MB_all_otu_table <- as.matrix(MB_all_tax_ps_rel@otu_table)
MB_all_otu_table <- sweep(MB_all_otu_table,1,rowMaxs(MB_all_otu_table),"/")
MB_all_tax_ps_rel <- merge_phyloseq(otu_table(MB_all_otu_table, taxa_are_rows = TRUE), MB_all_tax_ps_rel@sam_data)

write.csv(MB_all_otu_table, "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/MB_all_otu_table.csv")  
write.csv(MB_merged_otu_table, "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/MB_merged_otu_table.csv")  
```

```{r filter by Kendall correlations, include=FALSE, echo =FALSE}
bit=function(x){print(x[c(1:10),c(1:10)]); print(dim(x))}  #shows top-left corner of a data.frame/matrix; useful for large datasets
colMax <- function(data) sapply(data, max, na.rm = TRUE) #get column-wise maximum value
Col2RN<-function(df, x){row.names(df)<-df[,x]; df<-df[-x]; return(df)} #convert column to row.names

# a<-readRDS("/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MB_all_tax_ps_rel.Rdata")
# 
# otus<-as.data.frame(otu_table(a))
#   names(otus)
# meta<-as.data.frame(sample_data(a))
#   names(otus)<-meta$collection_date

otus<-MB_merged_otu_table
  
  #filter by sd or some other measure of variance
otus <- otus[specnumber(otus) > 1 , ]  #must occur in at least 2 time points, with zero abundance being a real data point in this case 

#compute CORRELATION after transposing dataset
corMB <- cor(t(otus), method = "kendall")

#save as dataframe and reshape; useful for plotting
corMB[lower.tri(corMB, T)] <- NA
corMB <- melt(corMB)%>%
  filter(!is.na(value))%>%
  filter(value>=0.7)

length(unique(unlist(corMB[,1:2])))  #339 unique taxa
    
    #save(corMB, file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/analysis/MBKendall.Rdata")
    
    #sanity check plot
    #plot(otus["f:Paracalanidae",]~otus["f:Thalassiosiraceae",])
   
unique_kend_taxa <- as.character(unique(unlist(corMB[,1:2])))
MB_merge_filt = prune_taxa(unique_kend_taxa,otu_table(MB_merged_otu_table, taxa_are_rows = TRUE))
MB_all_filt = prune_taxa(unique_kend_taxa,otu_table(MB_all_tax_ps_rel, taxa_are_rows = TRUE))
#write.csv(as.data.frame(MB_all_filt@.Data), "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/MB_all_filt.csv")
#write.csv(as.data.frame(MB_merge_filt@.Data), "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/MB_merge_filt.csv")
```

###Network analysis
```{r network analysis, echo=FALSE, message=FALSE, warning=FALSE}
options(stringsAsFactors = FALSE);
MB_Data <- as.data.frame(MB_merge_filt)
#dim(MB_Data);
#names(MB_Data);
datExpr0_MB = t(MB_Data) #transpose

#check if all taxa have good data (no zero variance datapoints and too many missing values)
gsg = goodSamplesGenes(datExpr0_MB, verbose = 3);
gsg$allOK

#run if not all taxa are ok!
if (!gsg$allOK)
{
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0) 
    printFlush(paste("Removing genes:", paste(names(datExpr0_MB)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0) 
    printFlush(paste("Removing samples:", paste(rownames(datExpr0_MB)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExpr0_MB = datExpr0_MB[gsg$goodSamples, gsg$goodGenes]
}

#make tree of samples to visualize ant obvious outliers
sampleTree_MB = hclust(tau.dist(datExpr0_MB));
#sizeGrWindow(12,9)
par(cex = 0.6);
#par(mar = c(0,4,2,0))
#pdf("/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/network/net_dendro_tau.pdf")
plot(sampleTree_MB, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5, 
     cex.axis = 1.5, cex.main = 2)
#dev.off()

# Plot a line to show the cut
#abline(h = 15, col = "red");
# Determine cluster under the line
clust_MB = cutreeStatic(sampleTree_MB, cutHeight = 15, minSize = 8)
table(clust_MB)
# clust 1 contains the samples we want to keep.
keepSamples = (clust_MB==1)
datExpr_MB = datExpr0_MB[keepSamples, ]
nGenes_MB = ncol(datExpr_MB)
nSamples_MB = nrow(datExpr_MB)

#load trait data
traitData_MB = read.csv("/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/metadata_MB_merged_sam.csv")
#dim(traitData_MB)
#names(traitData_MB)

# remove columns that hold information we do not need.
allTraits_MB = traitData_MB[, -c(1:17,19:20,21:32,34:46,50,53:64)];
dim(allTraits_MB)
names(allTraits_MB)

meta = as.data.frame(allTraits_MB)
meta = meta[, -c(2)]

# Form a data frame analogous to expression data that will hold the traits.
Samples_MB = rownames(datExpr_MB)
traitRows_MB = match(Samples_MB, allTraits_MB$SAMPLING_date);
rownames(allTraits_MB) = Samples_MB
datTraits_MB = allTraits_MB[,-c(2)]
collectGarbage();

sampleTree2_MB = hclust(tau.dist(datExpr_MB), method = "average")
# Convert traits to a color representation: white means low, red means high, grey means missing entry
traitColors_MB = numbers2colors(datTraits_MB, signed = FALSE);
# Plot the sample dendrogram and the colors underneath.
#pdf("/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/network/net_dendro_tau_w_env.pdf")
plotDendroAndColors(sampleTree2_MB, traitColors_MB,
                    groupLabels = names(datTraits_MB), 
                    main = "Sample dendrogram and trait heatmap")
#dev.off()
#STEP 2
allowWGCNAThreads()

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to=20, by=2))
# Call the network topology analysis function
sft_MB = pickSoftThreshold(datExpr_MB, powerVector = powers, verbose = 5)
# Plot the results:

#sizeGrWindow(9, 5)
#pdf("/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/network/pick_soft_threshold.pdf")
par(mfrow = c(1,2));
cex1 = 0.9;
# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft_MB$fitIndices[,1], -sign(sft_MB$fitIndices[,3])*sft_MB$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft_MB$fitIndices[,1], -sign(sft_MB$fitIndices[,3])*sft_MB$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# this line corresponds to using an R^2 cut-off of h
abline(h=0.60,col="red")
 # Mean connectivity as a function of the soft-thresholding power
plot(sft_MB$fitIndices[,1], sft_MB$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft_MB$fitIndices[,1], sft_MB$fitIndices[,5], labels=powers, cex=cex1,col="red")
#dev.off()

#create modules
#Choose power based on what the soft threshold results look like, ours are showing a scale free topography at around 8. See figs above.
net_MB = blockwiseModules(datExpr_MB, power = 7,
                       TOMType = "unsigned", minModuleSize = 30,
                       reassignThreshold = 0, mergeCutHeight = 0.30,
                       numericLabels = TRUE, corType ="pearson",pamRespectsDendro = FALSE,
                       saveTOMs = TRUE,
                       saveTOMFileBase = "MB_TOM", 
                       verbose = 3)
#?blockwiseModules
# open a graphics window
#sizeGrWindow(12, 9)
# Convert labels to colors for plotting
mergedColors_MB = labels2colors(net_MB$colors)
mergedColors_MB = gsub("yellow","#ffc425", gsub("blue","#00aedb", gsub("red", "#d11141", gsub("green","#00b159",gsub("brown","#f37735",gsub("turquoise","#8c8c8c",mergedColors_MB))))))
#Plot the dendrogram and the module colors underneath
#pdf("/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/network/taxa_dendro_modules.pdf")
plotDendroAndColors(net_MB$dendrograms[[1]], mergedColors_MB[net_MB$blockGenes[[1]]],
                    "Module colors",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)
#dev.off()

moduleLabels_MB = net_MB$colors
moduleColors_MB = labels2colors(net_MB$colors)
moduleColors_MB = gsub("yellow","#ffc425", gsub("blue","#00aedb", gsub("red", "#d11141", gsub("green","#00b159",gsub("brown","#f37735",gsub("turquoise","#8c8c8c",mergedColors_MB))))))
MEs_MB = net_MB$MEs;
geneTree_MB = net_MB$dendrograms[[1]];

#STEP 3
nGenes_MB = ncol(datExpr_MB);
nSamples_MB = nrow(datExpr_MB);
# Recalculate MEs with color labels
MEs0_MB = moduleEigengenes(datExpr_MB, moduleColors_MB)$eigengenes
MEs_MB = orderMEs(MEs0_MB)
colnames(MEs_MB) = c("red","blue", "dark grey","orange","green","yellow", "grey")
#colnames(MEs_MB) = c("dark grey","black","blue","green","brown","red","yellow","grey")

moduleTraitCor_MB = cor(MEs_MB, datTraits_MB, use = "p");
moduleTraitPvalue_MB = corPvalueStudent(moduleTraitCor_MB, nSamples_MB);

#sizeGrWindow(12,8)
# Will display correlations and their p-values
textMatrix_MB =  paste(signif(moduleTraitCor_MB, 2), "\n(",
                    signif(moduleTraitPvalue_MB, 1), ")", sep = "");
dim(textMatrix_MB) = dim(moduleTraitCor_MB)
# Display the correlation values within a heatmap plot
#pdf("/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/network/module_cor_envir.pdf")
par(mar = c(6, 8.5, 3, 3));
labeledHeatmap(Matrix = moduleTraitCor_MB,
               xLabels = names(datTraits_MB),
               yLabels = names(MEs_MB),
               yColorLabels = TRUE,
               ySymbols = names(MEs_MB),
               colorLabels = FALSE,
               colors = blueWhiteRed(50),
               textMatrix = textMatrix_MB,
               setStdMargins = FALSE,
               cex.text = 0.5,
               naColor = "grey",
               zlim = c(-1,1),
               main = paste("Module-trait relationships"))
#dev.off()


#STEP 5
modules = c("#00b159", "#ffc425", "#00aedb","#f37735","#8c8c8c", "#d11141");
# Select module probes
probes = names(as.data.frame(datExpr_MB))
inModule = is.finite(match(moduleColors_MB, modules));
modProbes = probes[inModule];
#modGenes = annot$gene_symbol[match(modProbes, annot$substanceBXH)];
# Select the corresponding Topological Overlap
TOM = TOMsimilarityFromExpr(datExpr_MB, power = 7);
modTOM = TOM[inModule, inModule];
dimnames(modTOM) = list(modProbes, modProbes)

# Export the network into edge and node list files Cytoscape can read
cyt = exportNetworkToCytoscape(modTOM,
                               edgeFile = paste("CytoscapeInput-edges-", paste(modules, collapse="-"), ".txt", sep=""),
                               nodeFile = paste("CytoscapeInput-nodes-", paste(modules, collapse="-"), ".txt", sep=""),
                               weighted = TRUE,
                               threshold = 0.1,
                               nodeNames = modProbes,
                               nodeAttr = moduleColors_MB[inModule])

write.csv(cyt$edgeData, file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/MB_edge.csv")
write.csv(cyt$nodeData, file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/MB_node.csv")
```

```{r network traits based on PCA,echo=FALSE,message=FALSE, warning=FALSE}
meta<-read.csv("/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/m2w_metadata_MB.csv")
#names(meta)
DATES<-str_split(meta$SAMPLING_date_time, " ", simplify=T)[,2]
meta<-meta[,c(18, 47,48, 51,52)]
standard<-decostand(meta, method="max") #standardize variables

	#PCA
envir.PCA <- princomp(standard)
summary(envir.PCA)
loadings(envir.PCA)
Loadings <- as.data.frame(envir.PCA$loadings[,1:2])
fviz_eig(envir.PCA)
#fviz_pca_ind(envir.PCA)
  
#plotting w ggfortify; see https://cran.r-project.org/web/packages/ggfortify/vignettes/plot_pca.html
#pdf("/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/envirPCA.pdf")
autoplot(envir.PCA,
         loadings = T, loadings.label = F, loadings.label.size = 4)+
  theme_minimal()+
  annotate("text", x = -0.2, y=0.28, label = "Upwelling", color = "red")+
  annotate("text", x = 0.25, y=0.21, label = "Nitrate", color = "red")+
  annotate("text", x = -0.05, y= -0.07, label = "Temperature", color = "red")+
  annotate("text", x = 0.032, y= -0.01, label = "Salinity", color = "red")+
  annotate("text", x = -0.06, y= -0.01, label = "Oxygen", color = "red")
#dev.off()
	

#moduleTraitCor_MB = cor(MEs_MB, datTraits_MB, use = "p");
#moduleTraitPvalue_MB = corPvalueStudent(moduleTraitCor_MB, nSamples_MB);

# Will display correlations and their p-values
#textMatrix_MB =  paste(signif(moduleTraitCor_MB, 2), "\n(",
#                    signif(moduleTraitPvalue_MB, 1), ")", sep = "");

#labeledHeatmap(Matrix = moduleTraitCor_MB,
#               xLabels = names(datTraits_MB),
#               yLabels = names(MEs_MB),
#               yColorLabels = TRUE,
#               ySymbols = names(MEs_MB),
#               colorLabels = FALSE,
#               colors = blueWhiteRed(50),
#               textMatrix = textMatrix_MB,
#               setStdMargins = FALSE,
#               cex.text = 0.5,
#               naColor = "grey",
#               zlim = c(-1,1),
#               main = paste("Module-trait relationships"))

```
###Pseudoautocorrelation of modules

```{r pseudo autocorrelation of all module MB,echo=FALSE,message=FALSE, warning=FALSE}
datExpr_MB <- as.data.frame(datExpr_MB)
blue_MB <- names(datExpr_MB)[moduleColors_MB=="#00aedb"] #blue
dg_MB <- names(datExpr_MB)[moduleColors_MB=="#8c8c8c"]#dark grey
orange_MB <- names(datExpr_MB)[moduleColors_MB=="#f37735"] #orange
green_MB <- names(datExpr_MB)[moduleColors_MB=="#00b159"] #green
yel_MB <-names(datExpr_MB)[moduleColors_MB=="#ffc425"] # yellow
red_MB <- names(datExpr_MB)[moduleColors_MB=="#d11141"] #red

#filter out desired color module
MB_taxa_yel = prune_taxa(yel_MB,MB_all_tax_ps_rel)
yel_otu_MB<-otu_table(MB_taxa_yel)
dates_MB<-ymd(sample_data(MB_taxa_yel)$collection_date)

MB_taxa_red = prune_taxa(red_MB,MB_all_tax_ps_rel)
red_otu_MB<-otu_table(MB_taxa_red)

MB_taxa_blue = prune_taxa(blue_MB,MB_all_tax_ps_rel)
blue_otu_MB<-otu_table(MB_taxa_blue)

MB_taxa_green = prune_taxa(green_MB,MB_all_tax_ps_rel)
green_otu_MB<-otu_table(MB_taxa_green)

MB_taxa_dg = prune_taxa(dg_MB,MB_all_tax_ps_rel)
dg_otu_MB<-otu_table(MB_taxa_dg)

dist.yel_sa_MB <- OneStep(OTUs=yel_otu_MB, DATES=dates_MB, METHOD="jaccard")
dist.red_sa_MB <- OneStep(OTUs=red_otu_MB, DATES=dates_MB, METHOD="jaccard")
dist.blue_sa_MB <- OneStep(OTUs=blue_otu_MB, DATES=dates_MB, METHOD="jaccard")
dist.green_sa_MB <- OneStep(OTUs=green_otu_MB, DATES=dates_MB, METHOD="jaccard")
dist.dg_sa_MB <- OneStep(OTUs=dg_otu_MB, DATES=dates_MB, METHOD="jaccard")

#pdf(file="/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/yel_black_PA.pdf",height=8,width=8)
#par(mfcol=c(2,1))
#par(mar=c(0.5,4,3,4))
boxplot(dist.yel_sa_MB, xaxt="n",ylim=c(0.2,1),at=c(1,3.5,7.5,12.5,18,19,22),main="MB",col="#ffc425", ylab="Jaccard Distance",xlim=c(-1.8,22))
#par(new=TRUE)
#plot(MB_all_tax_ps_rel@sam_data$diss_oxygen~dates_MB,type="l",ylim=c(5,7.5), lwd=5,xlab="", ylab="",yaxt="n",xaxt="n",col=alpha("grey",0.5))
#axis(side=4)
#mtext("dissolved oxygen",side=4,line=1.9,las=3)
#legend("topleft", "A", bty="n")
#mtext("Reference point 1",cex=0.8)
#par(mar=c(3,4,0.5,4))
#boxplot(dist.black_sa_MB, xlab="", ylim=c(0.2,1),at=c(1,3.5,7.5,12.5,18,19,22),ylab="Jaccard Distance", xlim=c(-1.8,22), ylim=c(0.3,1), col="black",xaxt="n")
#axis(side=1, at=c(1,3.5,7.5,12.5,18,19,22),
#     labels=c("July","Sep", "Dec","Apr", "Aug","Sep", "Dec"), las=3,cex.axis=0.7)
#par(new=TRUE)
#plot(MB_all_tax_ps_rel@sam_data$diss_oxygen~dates_MB,type="l",ylim=c(5,7.5), lwd=5,xlab="", ylab="",yaxt="n",xaxt="n",col=alpha("grey",0.5))
#axis(side=4)
#mtext("dissolved oxygen",side=4,line=1.9,las=3)
#legend("topleft", "B", bty="n")
#dev.off()

###BLUE AND RED
#pdf(file="/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/red_blue_PA.pdf",height=8,width=8)
env_MB <- read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/env_data/MBNMS_GHRSST_15_16.csv", header=TRUE)
par(mfcol=c(2,1))
par(mar=c(0.5,4,3,2))
boxplot(dist.red_sa_MB, xaxt="n",ylim=c(0.3,1),at=c(1,3.5,7.5,12.5,18,19,22),main="MB",col="#d11141",xlim=c(-1.8,22), ylab="Jaccard Distance")
#par(new=TRUE)
#plot(MB_all_tax_ps_rel@sam_data$temp~dates_MB,type="l",ylim=c(10,19), lwd=5,xlab="", ylab="",yaxt="n",xaxt="n",col=alpha("grey",0.5))
#axis(side=4)
#mtext("SST",side=4,line=1.9,las=3)
legend("topleft", "A", bty="n")
#mtext("Reference point 1",cex=0.8)
par(mar=c(3,4,0.5,2))
boxplot(dist.blue_sa_MB, xlab="", at=c(1,3.5,7.5,12.5,18,19,22),ylab="Jaccard Distance", xlim=c(-1.8,22), ylim=c(0.3,1), col="#00aedb",xaxt="n")
axis(side=1, at=c(1,3.5,7.5,12.5,18,19,22),
     labels=c("July","Sep", "Dec","Apr", "Aug","Sep", "Dec"), las=3,cex.axis=0.7)
#axis(side=1, at=c(1,2,3,4,5,6,7), labels=c("0","68","133","229","357","489","515"),las=1,cex.axis=1)
#par(new=TRUE)
#plot(MB_all_tax_ps_rel@sam_data$temp~dates_MB,type="l",ylim=c(10,19), lwd=5,xlab="", ylab="",yaxt="n",xaxt="n",col=alpha("grey",0.5))
#axis(side=4)
#mtext("SST",side=4,line=1.9,las=3)
legend("topleft", "B", bty="n")
#ev.off()


##### GREEN AND DARK GREY
#pdf(file="/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/green_dg_PA.pdf",height=8,width=8)
par(mfcol=c(2,1))
par(mar=c(0.5,4,3,2))
boxplot(dist.green_sa_MB, xaxt="n",ylim=c(0.3,1),at=c(1,3.5,7.5,12.5,18,19,22),main="MB",col="#00b159",xlim=c(-1.8,22), ylab="Jaccard Distance")
#par(new=TRUE)
#plot(MB_all_tax_ps_rel@sam_data$chlorophyll~dates_MB,type="l",ylim=c(0,19), lwd=5,xlab="", ylab="",yaxt="n",xaxt="n",col=alpha("grey",0.5))
#axis(side=4)
#mtext("Chlorophyll",side=4,line=1.9,las=3)
legend("topleft", "A", bty="n")
#mtext("Reference point 1",cex=0.8)
par(mar=c(3,4,0.5,2))
boxplot(dist.dg_sa_MB, xlab="", at=c(1,3.5,7.5,12.5,18,19,22),ylab="Jaccard Distance", xlim=c(-1.8,22), ylim=c(0.3,1), col="#8c8c8c",xaxt="n")
axis(side=1, at=c(1,3.5,7.5,12.5,18,19,22),
     labels=c("July","Sep", "Dec","Apr", "Aug","Sep", "Dec"), las=3,cex.axis=0.7)
#axis(side=1, at=c(1,2,3,4,5,6,7), labels=c("0","68","133","229","357","489","515"),las=1,cex.axis=1)
#par(new=TRUE)
#plot(MB_all_tax_ps_rel@sam_data$chlorophyll~dates_MB,type="l",ylim=c(0,19), lwd=5,xlab="", ylab="",yaxt="n",xaxt="n",col=alpha("grey",0.5))
#axis(side=4)
#mtext("Chlorophyll",side=4,line=1.9,las=3)
legend("topleft", "B", bty="n")
#dev.off()
```

```{r lag analysis, eval=FALSE, include=FALSE}
#LAGGED CORRELATION Matrix
source("/Volumes/GoogleDrive/Team Drives/m2w_analysis/analysis/LagAnalysis.R")
lagMB <- LAGCOR(otus, lag = 1)
    #reshape
    lagMB <- melt(lagMB)%>%
      filter(!is.na(value))%>%
      filter(value > 0.62)%>%
      arrange(desc(value))
        lagMB[,1] <- rownames(otus)[lagMB[,1]]
        lagMB[,2] <- rownames(otus)[lagMB[,2]]
        names(lagMB) <- c("Lagging", "Leading", "Kendall")
        
    head(lagMB)
    length(unique(lagMB$Lagging))

    #save(lagMB, file = "/Volumes/GoogleDrive/Team Drives/m2w_analysis/analysis/LaggedKendall.Rdata")
        
    #especially interesting leading/lagging correlations might be taxa that appear in at least 3-4 time points...
    #and with very high correlation coefficients
    lagMB[lagMB$Lagging %in% row.names(otus)[specnumber(otus)>3] &
      lagMB$Kendall > 0.85,]
  
```

### Supplementary Figures
Supplementary figures based to the Microbes to Mammals manuscript.
####Richness and Pseudoautocorrelation of whole community
```{r Richness of total taxa, echo=FALSE, fig.cap="\\label{fig:fig S1}Figure S1: A: Total richness (number of taxa) in Monterey Bay National Marine Sanctuary through April 2015 - December 2016 with sea surface temperature anomaly. B: Pseudo-autocorrelation analyses across all taxa compared to the time point prior to it. First time point is zero due to not having a reference point. The vertical dashed line represents December 31, 2015.", fig.height=8, fig.width=6.5}
#pdf(file="/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/Supplemental/obs_richness_PA.pdf",height=8,width=10)
par(mar=c(3,4,2,4.5))
par(mfrow=c(1,2))
otu_table(MB_all_filt) <- otu_table(round(as((otu_table(MB_all_filt)), "matrix")), taxa_are_rows(MB_all_filt)) #doesn't work to 
richness_total_MB <- setNames(cbind(estimate_richness(MB_all_filt, measures="Observed"),MB_all_tax_ps_rel@sam_data$collection_date, MB_all_tax_ps_rel@sam_data$locus),c("richness","YEAR_DATE", "locus"))
boxplot(richness_total_MB$richness~ymd(richness_total_MB$YEAR_DATE),
        main="", at=c(1,4,7,9,13,17,18,21), cex.axis = 0.8,xlim=c(0,21), las=2, ylim=c(0,100),
       xaxt="n",col="grey90",ylab=c("Number of taxa"))
abline(v=c(10), col=c("black"), lty=c(2), lwd=c(1))
axis(side=1, at=c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21),
 labels=c("Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec", "Jan", "Feb","Mar",
            "Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec"), las=3,cex.axis=0.7)
mtext("2015", side=1, line=3, at=4,cex=0.8)
mtext("2016", side=1, line=3, at=16,cex=0.8)
env_MB <- read.csv("/Users/anni_djurhuus/Documents/Projects/MBON/m2w/env_data/MBNMS_GHRSST_15_16.csv", header=TRUE)
par(new=TRUE)
plot(env_MB$Mean.SST~dmy(env_MB$Date),type="l",ylim=c(10,20), lwd=5,xlab="", ylab="",yaxt="n",xaxt="n",col=alpha("grey",0.5))
axis(side=4)
mtext("Sea surface temperature",side=4,line=1.9,las=3)
legend("topleft", "A", bty="n")

#PA
OTUs_MB<-otu_table(MB_all_tax_ps_ts)
dates_MB<-ymd(sample_data(MB_all_tax_ps_ts)$collection_date)
dist.res_MB_sa <- OneStep(OTUs=OTUs_MB, DATES=dates_MB, METHOD="jaccard")
#plot object in base R
par(mar=c(3,4.5,2,4))
boxplot(dist.res_MB_sa, at=c(4,7,9,13,17,18,21),xlim=c(0,21),xlab="", ylab="Euclidian Distance", xlim=c(0,21.5), ylim=c(0.7,0.9), col="grey90",xaxt="n")
abline(v=c(10), col=c("black"), lty=c(2), lwd=c(1))
axis(side=1, at=c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21),
     labels=c("Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec", "Jan", "Feb","Mar",
              "Apr", "May", "June", "July", "Aug","Sep", "Oct", "Nov", "Dec"), las=3,cex.axis=0.7)
mtext("2015", side=1, line=3, at=4,cex=0.8)
mtext("2016", side=1, line=3, at=16,cex=0.8)
legend("topleft", "B", bty="n")
#dev.off()
```
####Heatmap and dendrogram of top taxa
```{r heatmap of top taxa, echo=FALSE, fig.cap="\\label{fig:fig S2}Figure S2: Dendrogram of the top 100 taxa for FK left and MB right. Dissimilarity between taxa was calculated based on a Euclidean distance matrix.", fig.height=8, fig.width=10, message=FALSE, warning=FALSE}
#pdf(file="/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/Supplemental/heatmap.pdf",height=10,width=7)
top_50_MB <- prune_taxa(names(sort(taxa_sums(MB_merge_filt),
         TRUE)[1:100]),MB_merge_filt)

#pushViewport(viewport(x = 0))
draw(Heatmap(data.frame(top_50_MB@.Data),row_names_gp = gpar(fontsize = 8),col=pal,column_title_gp = gpar(fontsize = 8),clustering_distance_rows = "euclidean",column_dend_height = unit(6, "mm"),column_names_gp = gpar(fontsize = 8),row_title_gp=gpar(fontsize=8),km_title = "cluster% i", name = "MB"),k=4, newpage = FALSE)
#popViewport()
#dev.off()
```
####Line plots (a version of this will probably go into main document)
```{r line plots, echo=FALSE, fig.cap="\\label{fig:fig 3}Figure 3: Some of the co-occuring organisms with the highest correlations from FKNMS (left) and MBNS (right).", fig.height=5.5, fig.width=5.5, message=FALSE, warning=FALSE}
#FK_all_merge$collection_date <- ymd(FK_all_merge$collection_date)
#FK_all_merge <- arrange(FK_all_merge, collection_date)
#pdf(file="/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/Supplemental/line_plots.pdf",height=10,width=7)
par(mfcol=c(2,1))
par(mar=c(1,4,3,2))
#MB
MB_all_merge$collection_date <- ymd(rownames(MB_all_merge))
MB_all_merge <- arrange(MB_all_merge, collection_date)
plot(MB_all_merge$`f:Noelaerhabdaceae`~MB_all_merge$collection_date,col="#E69F00",type="l", pch=16,lwd=2,ylab=c(""), xlab="",xaxt="n",yaxt="n",ylim=c(0,1.2))
#lines(MB_all_merge$`f:Pelagibacteraceae`~MB_all_merge$collection_date, col="gold", pch=16, lwd=2)
lines(MB_all_merge$`f:Engraulidae`~MB_all_merge$collection_date,lwd=2, col="darkorange3")
lines(MB_all_merge$`f:Balaenopteridae`~MB_all_merge$collection_date,lwd=2, 
col="black")
#lines(MB_all_merge$f.Calanidae/max(MB_all_merge$f.Calanidae)~MB_all_merge$collection_date,lwd=2, col="grey")
legend("topleft",legend=c("Noelaerhab.", "Engraulidae","Balaenoptera"),ncol=2,col=c("darkorange3","black"), bty="n", lty=c(1,1,1), cex=0.6)
abline(v=c(16805), col=c("black"), lty=c(2), lwd=c(1))
par(mar=c(4,4,0,2))
plot(MB_all_merge$`f:Paracalanidae`~MB_all_merge$collection_date, col="darkolivegreen3", pch=16, lwd=2,xlab="", xaxt="n",ylab=c(""),yaxt="n",ylim=c(0,1.2),type="l")
lines(MB_all_merge$`f:Thalassiosiraceae`~MB_all_merge$collection_date, col="#009E73", pch=16, lwd=2)
lines(MB_all_merge$`f:Chaetocerotaceae`~MB_all_merge$collection_date, col="darkolivegreen", pch=16, lwd=2)
lines(MB_all_merge$`f:Clupeidae`~MB_all_merge$collection_date,lwd=2, col="grey")
axis(1, ymd(MB_all_merge$collection_date), format(ymd(MB_all_merge$collection_date), "%b"), cex.axis = .7, las=3)
legend("topleft",legend=c("Paracalanidae", "Thalassiosir.","Chaetoceros", "Clupeidae"),col=c("darkolivegreen3","#009E73","darkolivegreen", "grey"), bty="n", ncol=2,lty=c(1,1,1), cex=0.6)
abline(v=c(16805), col=c("black"), lty=c(2), lwd=c(1))
mtext("2015", side=1, line=3, at=15650,cex=0.7)
mtext("2016", side=1, line=3, at=19000,cex=0.7)

#dev.off()
#cor(MB_all_merge$f.Paracalanidae,MB_all_merge$f.Calanidae)

```
####Correlation matrix of top taxa
```{r plots of correlations, echo=FALSE, fig.cap="\\label{fig:fig S4}Figure S4: Correlation of the top 50 taxa for Monterey Bay only including significant values of a p-value equal to or less than 0.05. These correlation matrices were performed on relative abundance indices and ordered by hierarcical clustering on dissimilarities between taxa", fig.height=8, fig.width=7, message=FALSE, warning=FALSE}
#MB
#pdf(file="/Volumes/GoogleDrive/Team Drives/m2w_analysis/plots/Supplemental/correlation_top50.pdf",height=8,width=7)
top_taxa_merge_MB <- prune_taxa(names(sort(taxa_sums(MB_all_tax_ps_ts_merge),
         TRUE)[1:50]),MB_all_tax_ps_ts_merge) #use transformed merged data
cor_MB <- as.matrix(top_taxa_merge_MB@otu_table)
#res2_MB<-rcorr(cor_MB,type="spearman")
res2_MB <- cor(cor_MB, method="kendall")
#saveRDS(res2_MB, "/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MEGAN6/post_decontam/MB/taxa_cor_MB.Rdata")  
corrplot(res2_MB, type="lower", order="hclust", 
          sig.level = 0.05, tl.srt=45,tl.col="black", insig = "blank",tl.cex=0.6,diag=FALSE)
#dev.off()
```

```{r Ordinations, eval=FALSE, message=FALSE, warning=FALSE, include=FALSE}
MB_all_tax_df <- t(data.frame(MB_all_tax_ps_rel@otu_table))
vegdist_MB <- vegdist(MB_all_tax_df, method="jaccard") # Bray-Curtis distances between the rows
metaMDS_MB <- metaMDS(vegdist_MB,verbose = FALSE) # k is the number of dim
#metaMDS_MB # view results
# plot solution 
x_MB <- metaMDS_MB$points[,1]
y_MB <- metaMDS_MB$points[,2]

FK_all_tax_df <- t(data.frame(FK_all_tax_ps_rel@otu_table))
vegdist_FK <- vegdist(FK_all_tax_df, method="jaccard") # Bray-Curtis distances between the rows
metaMDS_FK <- metaMDS(vegdist_FK,verbose = FALSE) # k is the number of dim
# plot solution phytoplankton
x_FK <- metaMDS_FK$points[,1]
y_FK <- metaMDS_FK$points[,2]

all_data <- merge_phyloseq(MB_all_tax_ps_ts,FK_all_tax_ps_ts)
all_Data.ord <- ordinate(all_data, "NMDS", "jaccard")
```

```{r plot ordinations, eval=FALSE, fig.cap="\\label{fig:fig 5}Figure 5: Canonical correspondence analysis of both locations FK (left) and MB (right). The points are colored by replicate with point type representing year (2015=square, 2016=circle). The text is sample collection date. Distance matrix is jaccard.", fig.height=4, fig.width=6, message=FALSE, warning=FALSE, include=FALSE}
par(mfrow=c(1,2))
par(mar=c(4,5,2,0))
sam_FK_df <- data.frame(FK_all_tax_ps_rel@sam_data)
plot(x_FK, y_FK,xlab="Coordinate 1", main="", ylab="Coordinate 2",col=c(cbbPalette[FK_all_tax_ps_rel@sam_data$collection_date]),pch=c(15,19)[as.factor(FK_all_tax_ps_rel@sam_data$year)])
text(x_FK, y_FK,label=FK_all_tax_ps_rel@sam_data$collection_date,cex=0.5)
expl_FK <- subset(sam_FK_df, select=c("TMP_C","CHL_MG_M.3","NO3_um"))
plot(envfit(metaMDS_FK, expl_FK),col="black", cex=0.5)
legend("topleft", "A", bty="n")

par(mar=c(4,1,2,4))
plot(x_MB, y_MB, xlab="Coordinate 1", main="", yaxt="n",ylab="Coordinate 2",col=c(cbbPalette[MB_18S_fam@sam_data$collection_date]),pch=c(15,19)[as.factor(MB_18S_fam@sam_data$year)])
text(x_MB, y_MB,label=MB_18S_fam@sam_data$collection_date,cex=0.5)
sam_MB_df <- data.frame(MB_18S_fam@sam_data)
expl_MB <- subset(sam_MB_df, select=c("temp","nitrate","salinity","chlorophyll"))
plot(envfit(metaMDS_MB, expl_MB),col="black",cex=0.5)
legend("topleft", "B", bty="n")
```
####Finding Kendall threshold for taxa filtering
```{r Finding Kendall threshold for merged otu table, echo=FALSE, message=FALSE, warning=FALSE}
#One can imagine different permutation procedures; here are two.

  #function to scramble an existing dataset
  PERMFUN <- function(z) {matrix(sample(z),nrow=nrow(z))}
  propObs <- function(data, x) {sum(abs(data) >= x) / length(data)}  #fraction in dataset greater than a given value
  
  #permutes non-zero numbers in data; different permutation method
  NEWPERM<-function(x){temp<-x 
                       temp[which(temp!=0|temp!=1)]<-sample(nonzero)
                       temp}  

#ObservedData
  ###Empirical data, for comparison
  otus<-MB_merged_otu_table
    #readRDS("/Volumes/GoogleDrive/Team Drives/m2w_analysis/data/MB_merged_otu_table.Rdata")
  #filter by sd or some other measure of variance
  otus <- otus[specnumber(otus) > 1 , ]  #must occur in at least 2 time points, with zero abundance being a real data point in this case 
  nonzero<-otus[which(otus!=0|otus!=1)] #index of non-zero and non-one numbers

  #compute CORRELATION after transposing dataset
  corMB <- cor(t(otus), method = "kendall")
  
  #save as vector, leaving out diagonal
  corMB.vector <- corMB[upper.tri(corMB)]
  
  #save as dataframe and reshape; useful for plotting
  corMB[lower.tri(corMB, T)] <- NA
  corMB <- melt(corMB)%>%
    filter(!is.na(value))
  
#Permutations  
  #Generate scrambled datasets;
  #Note the difference between permuting columns or rows (which just changes their order, and which doesn't do anything in this case, since these are comparisons of pairwise ranked values... the resulting distribution is the same (!) for every such permutation. Here, we're scrambling the contents of the data matrix itself.)

  set.seed(49)
  nPerms = 100
  CorPerms <- lapply(1:nPerms, FUN = function(x){
    PERMFUN(otus)%>%
      as.data.frame%>%
      filter(specnumber(.)>1)%>%
      t%>%
      cor(method = "kendall")%>%
      .[upper.tri(.)]
  })
  beep()

  # #visualize w eCDFs
  #       plot(ecdf(abs(CorPerms[[1]])), col = "lightgrey", xlim=c(0.5,1), ylim=c(0.85, 1))
  #         lapply(1:100, function(x) plot(ecdf(abs(CorPerms[[x]])), add = T, col = "lightgrey"))
  #             plot(ecdf(abs(corMB.vector)), col = "red", add=T)

#Benjamani-Hochberg_FDR
    y<-sapply(1:100, function(x) unlist(CorPerms[[x]]))
                      y<-unlist(y)
                      
                adj.p <- sapply(seq(0.5, 1, 0.005), function(tau) {p.adjust(propObs(y, tau), method = "fdr", n = length(corMB))})     
                
                tmp <- data.frame(
                  Tau = seq(0.5, 1, 0.005),
                  Adjusted.P=adj.p
                )
                
                ggplot(data=tmp, aes(y=Adjusted.P, x=Tau))+
                  geom_smooth()+
                  geom_point()+
                  geom_vline(aes(xintercept = 0.70), color = "red", alpha = 0.5)+
                  geom_hline(aes(yintercept = 0.05), color = "red", alpha = 0.5)
                

```
